### PR TITLE
feat(socket): berkeley socket API v2

### PR DESCRIPTION
**Pull request by robberphex**

### CODE REVIEW

**Comment by robberphex** [comment]
I noticed that there is a network API implementation https://github.com/bytecodealliance/wasm-micro-runtime/pull/836 .
@lum1n0us Could you check this PR?

**Comment by lum1n0us** [comment]
@robberphex  👍

**Comment by linclark** [comment]
Thanks for looking into this, @robberphex 

Unfortunately while the previous PR was dormant, WASI as a whole has been in a process of transition. The original `witx` format was intended as a stop-gap until Interface Types were more fully developed. We've now developed the [`wit`](https://github.com/bytecodealliance/wit-bindgen/blob/main/WIT.md) format which we are transitioning to. You can see more about this in the [notes from the last WASI meeting](https://github.com/WebAssembly/meetings/blob/main/wasi/2022/WASI-01-13.md#plan-for-the-transition-from-current-witx-based-apis-to-wit-and-canonical-abi), and I plan to update the repo (based on #451) this week.

Given that, the next steps for this to move forward would be having a champion step forward to move the proposal through the Phase process, moving it to its own repo and rewriting it in the `wit` format.

**Comment by badeend** [comment]
To those who are interested; I've created a draft proposal at: https://github.com/badeend/WASI-Networking

File: phases/snapshot/docs.md [hunk 1]
```diff
@@ -2011,7 +2006,7 @@ The path at which to create the directory.
 
 #### <a href="#path_filestat_get" name="path_filestat_get"></a> `path_filestat_get(fd: fd, flags: lookupflags, path: string) -> Result<filestat, errno>`
 Return the attributes of a file or directory.
-Note: This is similar to `fstatat` in POSIX.
+Note: This is similar to `stat` in POSIX.
```
Comment (appcypher) [inline phases/snapshot/docs.md:2009]:
```suggestion
Note: This is similar to `fstat` in POSIX.
```

**Pull request closed without merging**